### PR TITLE
pdk 2.6.1 fixes pdk-validate action

### PR DESCRIPTION
**Pull request by andylytical**
Changes:

- fixes the "pdk-validate" action ( as per https://github.com/puppetlabs/pdk-docker/issues/43 )

- bump actions/checkout@v3 (as per https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/ )

- fixes for pdk validate

### CODE REVIEW

**Review by billglick — COMMENTED** [review]
A few questions and suggestions.

File: .github/workflows/pdk-validate.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: "ubuntu-latest"
     steps:
       - name: "Clone repository"
-        uses: "actions/checkout@v2"
+        uses: "actions/checkout@v3"
```
[the remark this answers is not in the document]
Comment (andylytical) [reply .github/workflows/pdk-validate.yml:11]: Resolves the warning [Node.js 12 actions are deprecated](https://github.com/ncsa/puppet-profile_firewall/actions/runs/4139225805).

File: lib/facter/custom_firewallchains.rb [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
   # https://puppet.com/docs/puppet/latest/fact_overview.html
   setcode do
-    data = Hash.new
-    for table in [ 'nat', 'filter' ] do
-      lines = Facter::Core::Execution.execute( "/usr/sbin/iptables -t #{table} -S" )
-      chain_names = lines.scan(/^-N (.+)$/).flatten
+    data = {}
+    [ 'nat', 'filter' ].each do |table|
+      lines = Facter::Core::Execution.execute("/usr/sbin/iptables -t #{table} -S")
+      chain_names = lines.scan('/^-N (.+)$/').flatten
```
[the remark this answers is not in the document]
Comment (andylytical) [reply lib/facter/custom_firewallchains.rb:9]: Resolves `pdk validate` recommendations that weren't present in the output from pdk 2.5.0.

It is mentioned in the log as well as the PR ... `fixes for pdk validate`

File: metadata.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
       "version_requirement": ">= 4.10.0 < 7.0.0"
     }
   ],
-  "pdk-version": "2.5.0",
-  "template-url": "pdk-default#2.5.0",
-  "template-ref": "tags/2.5.0-0-g369d483"
+  "pdk-version": "2.6.1",
+  "template-url": "pdk-default#2.7.1",
+  "template-ref": "tags/2.7.1-0-g9a16c87"
```
[the remark this answers is not in the document]
Comment (andylytical) [reply metadata.json:45]: Appears that PDK is modular and various parts have different version numbers.

Templates is currently version 2.7.1

However, I see that PDK version is (correctly) 2.6.1, which makes my commit log incorrect. I'll make a new log message with the correct version number.

[the remark this answers is not in the document]
Comment (andylytical) [reply metadata.json:45]: I made a new commit with corrected log info. Also updated this PR title.

Also added URLs to the commit log message for the various issues.

**Review by billglick — APPROVED** [review]

**Pull request closed without merging**